### PR TITLE
DCA-110: improve reading yml config file

### PR DESCRIPTION
**Pull request by mmizin**

### CODE REVIEW

File: app/util/conf.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        obj = read_yml_file(JIRA_YML)
+        self.hostname = obj['settings']['env']['application_hostname']
+        self.protocol = obj['settings']['env']['application_protocol']
+        self.port = obj['settings']['env']['application_port']
+        self.postfix = obj['settings']['env']['application_postfix'] or ""
+        self.admin_login = obj['settings']['env']['admin_login']
+        self.admin_password = obj['settings']['env']['admin_password']
+        self.concurrency = obj['settings']['env']['concurrency']
+
+    def __repr__(self):
```
Comment (ometelytsia) [inline app/util/conf.py]: why do we need this method?

File: app/util/conf.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return f'hostname = {self.hostname!r}\n' \
+               f'protocol = {self.protocol!r}\n' \
+               f'port = {self.port!r}\n' \
+               f'postfix = {self.postfix!r}\n' \
+               f'admin_login = {self.admin_login!r}\n' \
+               f'admin_password = {self.admin_password!r}\n' \
+               f'concurrency = {self.concurrency!r}\n'
+
+    def get_server_url(self):
```
Comment (ometelytsia) [inline app/util/conf.py]: what do you think about converting this method to a property?

File: app/util/conf.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        return f'{self.protocol}://' \
```
Comment (ometelytsia) [inline app/util/conf.py]: can we have a one-liner here

File: app/util/conf.py (hunk 1 above)
Comment (mmizin) [reply app/util/conf.py]: It's like a best practice. It can be also useful for debugging.

File: app/util/conf.py (hunk 2 above)
Comment (mmizin) [reply app/util/conf.py]: Yes, it's a good idea !!! Thanks for the suggestion )

File: app/util/conf.py (hunk 3 above)
Comment (mmizin) [reply app/util/conf.py]: Yes, of course, I've done it for the best readability. But can change it!

Comment (mmizin) [reply app/util/conf.py]: Now it looks even better !!!

File: app/selenium_ui/conftest.py [hunk 4]
```diff
@@ -46,17 +47,6 @@
         pass
 
 
-def application_url():
-    with JIRA_YML.open(mode='r') as fs:
-        jira_yaml = yaml.load(fs, Loader=yaml.FullLoader)
```
[the remark this answers is not in the document]
Comment (mmizin) [reply app/selenium_ui/conftest.py]: done!

File: app/selenium_ui/jira/modules.py [hunk 5]
```diff
@@ -10,12 +10,13 @@
 from selenium.webdriver.support.ui import Select
```
[the remark this answers is not in the document]
Comment (mmizin) [reply app/selenium_ui/jira/modules.py]: done!

File: app/selenium_ui/conftest.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import atexit
 import csv
 import datetime
+import glob
```
Comment (ometelytsia) [inline app/selenium_ui/conftest.py]: why do you need this package?

File: app/selenium_ui/jira/modules.py [hunk 7]
```diff
@@ -130,7 +128,7 @@ def __filer_epic(element):
             webdriver.find_element_by_id(ISSUE_TYPE_DROPDOWN).click()
             issue_elements_in_dropdown = webdriver.find_elements_by_class_name("aui-list-item")
             if issue_elements_in_dropdown:
-                filtered_issue_elements = list(filter(__filer_epic, issue_elements))
+                filtered_issue_elements = list(filter(__filer_epic, issue_elements_in_dropdown))
```
Comment (ometelytsia) [inline app/selenium_ui/jira/modules.py]: I believe this was fixed in another PR.
Just merge with dev to get this fix, but not make it here.

File: app/selenium_ui/conftest.py (hunk 6 above)
Comment (mmizin) [reply app/selenium_ui/conftest.py]: I thought it was before.
I will remove it

File: app/util/data_preparation/confluence/prepare-data.py [hunk 8]
```diff
@@ -39,13 +33,13 @@ def write_to_file(file_name, list):
 def main():
     print("Started preparing data")
 
-    url = sys.argv[1]
+    url = CONFLUENCE_SETTINGS.server_url
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence/prepare-data.py]: If we are going with this approach.
1. We need to change prepare_data.py call in jira.yml:
from
python util/data_preparation/jira/prepare-data.py `${application_protocol}://${application_hostname}:${application_port}${application_postfix}`
to
`python util/data_preparation/jira/prepare-data.py`
2. Prepate date should fail gracefully if jira.yml does not have correct values for host, port, ect. With explanation of what and where has to be set.

File: app/selenium_ui/jira/modules.py (hunk 7 above)
Comment (mmizin) [reply app/selenium_ui/jira/modules.py]: I found out that I've not changed this variable.

**Pull request closed without merging**